### PR TITLE
Test Persistent Object with keys

### PR DESCRIPTION
**Pull request by pascal-brand-st-dev**
xtest 1014 is introduced to test TEE_CreatePersistentObject() associated
with a key.

Suggested-by: Miraje Gentilal <miraje.gentilal@hotmail.com>
Signed-off-by: Pascal Brand <pascal.brand@st.com>

### CODE REVIEW

**Comment by pascal-brand-st-dev** [comment]
Depends on https://github.com/OP-TEE/optee_os/pull/795

File: host/xtest/xtest_1000.c [hunk 1]
```diff
@@ -164,6 +165,17 @@ ADBG_CASE_DEFINE(XTEST_TEE_1013, xtest_tee_test_1013,
 		"Description of how to implement ..."
 		 );
 
+ADBG_CASE_DEFINE(XTEST_TEE_1014, xtest_tee_test_1014,
```
Comment (jenswi-linaro) [inline host/xtest/xtest_1000.c]: Wouldn't this belong in the 6000 range or possibly the 20000 range?

**Comment by pascal-brand-st-dev** [comment]
updated

File: ta/storage/storage.c [hunk 2]
```diff
@@ -270,3 +271,99 @@ TEE_Result ta_storage_cmd_next_enum(uint32_t param_types, TEE_Param params[4])
...
+
+	result = TEE_AllocateTransientObject(TEE_TYPE_AES, key_size,
+					     &transient_key);
+	if (result != TEE_SUCCESS) {
+		EMSG("Failed to Allocate transient object handle : 0x%x",
+		     result);
+		goto cleanup1;
+	}
+
+	result = TEE_GenerateKey(transient_key, key_size, (TEE_Attribute *)NULL,
```
Comment (jenswi-linaro) [inline ta/storage/storage.c]: Cast shouldn't be needed.

File: ta/storage/storage.c [hunk 3]
```diff
@@ -270,3 +271,99 @@ TEE_Result ta_storage_cmd_next_enum(uint32_t param_types, TEE_Param params[4])
...
+
+	result = TEE_OpenPersistentObject(TEE_STORAGE_PRIVATE,
+					  &objectID, sizeof(objectID),
+					  flags, &key);
+	if (result != TEE_SUCCESS) {
+		EMSG("Failed to open persistent key: 0x%x", result);
+		goto cleanup2;
+	}
+
+	TEE_GetObjectInfo(key, &keyInfo3);
```
Comment (jenswi-linaro) [inline ta/storage/storage.c]: We could perhaps compare the critical elements of `keyInfo2` with the ones in `keyInfo3` to see that they are preserved as expected.

**Comment by jenswi-linaro** [comment]
With my comments addressed:
`Reviewed-by: Jens Wiklander <jens.wiklander@linaro.org>`

**Pull request merged**